File: src/test/java/etlmail/context/PropertyBuilderTest.java
package etlmail.context;

import java.util.Properties;

import org.fest.assertions.Assertions;
import org.junit.Test;

import etlmail.context.PropertyBuilder;

public class PropertyBuilderTest {
    @Test
    public void avecUnePaireClefValeur() {
	final Properties resultat = new PropertyBuilder() //
		.key("clef").yields("valeur") //
		.asProperties();

	Assertions.assertThat(resultat.getProperty("clef")).isEqualTo("valeur");
    }
}

File: src/test/scala/etlmail/context/PropertyBuilderTest.scala
package etlmail.context

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.matchers.ShouldMatchers
import org.scalatest.FlatSpec

@RunWith(classOf[JUnitRunner])
class PropertyBuilderTest extends FlatSpec with ShouldMatchers {
  "one key-value pair" should "result in a properties object with one entry" in {
    // given
    val resultat = new PropertyBuilder() //
      .key("clef").yields("valeur") //
      .asProperties

    // then
    resultat should have size (1)
    resultat.getProperty("clef") should be("valeur")
  }
}
